R/constructors.R

```diff
@@ -141,6 +142,13 @@ new_quant_param <- function(type = c("double", "integer"),
     }
   }
 
+  # if (!is_unknown(default)) {
```

Comment:
The current signature for `default` is `default = unknown()`, but when you deprecate an argument you use `default = deprecated()` where `deprecated()` is from lifecycle (so you will import that function). And then your if statement is `if (lifecycle::is_present(default)) { warning }`